Modules/Scripted/DataProbe/DataProbe.py

```diff
@@ -328,7 +328,21 @@ def generateLayerName(self, slicerLayerLogic):
 
     def generateIJKPixelDescription(self, ijk, slicerLayerLogic):
         volumeNode = slicerLayerLogic.GetVolumeNode()
-        return f"({ijk[0]:3d}, {ijk[1]:3d}, {ijk[2]:3d})" if volumeNode else ""
+        if not volumeNode:
+            return ""
+        description = f"({ijk[0]:3d}, {ijk[1]:3d}, {ijk[2]:3d})"
+        instanceNumbersAttr = volumeNode.GetAttribute("DICOM.instanceNumbers")
+        if instanceNumbersAttr:
+            try:
+                # The DICOMScalarVolumePlugin ensures that the instance numbers are sorted
+                # so that the i-th slice in the volume node corresponds to the i-th value in the DICOM.instanceNumbers attribute.
+                instanceNumbers = [int(x) for x in instanceNumbersAttr.split(" ")]
```

Comment:
1. use `maxsplit` argument to prevent unnecessary computation (beyond `k` value) in `string.split(separator, maxsplit)`
2. do not use `int(x)` on all values, because it will fail if there are any unknown values.